tests/setup.py

```diff
@@ -119,37 +112,28 @@ def setUp(self) -> 'Generator':
         self.assertTrue(window)
         filename = expand(join("$packages", "LSP", "tests", "{}.txt".format(test_name)), window)
         self.config.init_options["serverResponse"] = server_capabilities
-        window.run_command("close_all")
-        # Cleanly shut down all window managers by ending all their sessions
-        for wm in windows._windows.values():
-            wm.end_sessions()
-
-            def condition() -> bool:
-                nonlocal wm
-                return len(wm._sessions) == 0
-
-            yield {"condition": condition, "timeout": TIMEOUT_TIME, "period": PERIOD_TIME}
-        # The Sublime windows are actually still valid, so WindowManagers don't remove themselves from this global
-        # dictionary. We do it by hand.
-        windows._windows.clear()
-        client_configs.all.clear()
         add_config(self.config)
-        self.wm = windows.lookup(window)  # create just this single one for the test
-        self.assertEqual(len(self.wm._sessions), 0)
+        self.wm = windows.lookup(window)
+        self.wm._configs.all.append(self.config)
         self.view = window.open_file(filename)
         yield {"condition": lambda: not self.view.is_loading(), "timeout": TIMEOUT_TIME, "period": PERIOD_TIME}
         self.assertTrue(self.wm._configs.syntax_supported(self.view))
         self.init_view_settings()
-        yield lambda: len(self.wm._sessions) > 0
-        sessions = self.wm._sessions.get(self.config.name, [])
-        self.assertEqual(len(sessions), 1)
-        self.session = sessions[0]
+        found_document_sync_listener = False
+        for listener in view_event_listeners[self.view.id()]:
+            if isinstance(listener, DocumentSyncListener):
+                # Bug in ST3?
```

Comment:
We're probably gonna remember this problem well for the rest of our lives but some detailed comment might be nice anyway.

Reply:
Basically my interpretation of the problem is that CI runs with ST window not in focus and that makes ST3 not trigger some events like on_load_async, on_activated, on_deactivated. That makes things not properly initialized on opening file (manager missing in DocumentSyncListener).

Reply:
Thanks, I've added your sentences as a comment.